decidim-core/spec/types/reportable_type_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+require "decidim/api/test"
+
+module Decidim
+  module Core
+    describe ReportableType do
+      include_context "with a graphql class type"
+      let!(:model) { create(:report, details: "Testing reason", locale: "en") }
+
+      include_examples "timestamps interface"
+
+      describe "id" do
+        let(:query) { "{ id }" }
+
+        it "returns the moderation's id" do
+          expect(response).to eq("id" => model.id.to_s)
+        end
+      end
+
+      describe "reason" do
+        let(:query) { "{ reason }" }
+
+        it "returns the moderation's reason" do
+          expect(response["reason"]).to eq(model.reason)
+        end
+      end
+
+      describe "details" do
+        let(:query) { "{ details }" }
+
+        it "returns the moderation's details" do
+          expect(response["details"]).to eq(model.details)
+        end
+      end
+
+      describe "locale" do
+        let(:query) { "{ locale }" }
+
+        it "returns the moderation's locale" do
+          expect(response["locale"]).to eq(model.locale)
+        end
+      end
+
+      describe "user" do
+        let(:query) { "{ user { id } }" }
+
+        it "returns the moderation's user" do
+          expect(response["user"]["id"]).to eq(model.user.id.to_s)
+        end
+
+        context "when the user is anonymous" do
```

Comment:
Maybe it is too early on Monday morning 😄 , but what does it means that the user is anonymous?